lerobot/scripts/control_robot.py

```diff
@@ -639,12 +701,16 @@ def on_press(key):
                 if not is_headless():
                     listener.stop()
 
-                logging.info("Waiting for threads writing the images on disk to terminate...")
-                for _ in tqdm.tqdm(
-                    concurrent.futures.as_completed(futures), total=len(futures), desc="Writting images"
-                ):
-                    pass
-                break
+                if has_camera > 0:
+                    logging.info("Waiting for subprocess writing the images on disk to terminate...")
+                    stop_workers(image_workers, image_queue)
+
+    except Exception:
+        traceback.print_exc()
+
+        if has_camera > 0:
+            logging.info("Waiting for subprocess writing the images on disk to terminate...")
+            stop_workers(image_workers, image_queue)
```

Comment:
TODO:(rcadene, aliberts): Not sure how to deal with this try and except because of `@safe_disconnect`